react/features/chat/replacement.js

```diff
@@ -1,5 +1,22 @@
 import { regexes } from './smileys';
 
+/* eslint-disable no-useless-escape, max-len */
+const replacePatterns = {
```

Comment:
I think I'd do something else here. Make this an array of replacement **functions**. A function can either returna replacement value or `undefined`. If `undefined` is returned we move on to the next function.

The last function would be "the identity function", which just returns the argument that was passed to it.

This way we can just iterate all the pattern functions and take the result without even looking at it, since we know it will be "something". WDYT?

Reply:
Ok, yep, this way it simplifies things will try it out.

Reply:
@saghul On a second thought, this actually will bring a lot of code duplication. All this with the string length check need to be in every function i order to decide whether to return the string or undefined. WDYT?

Reply:
I think the current code minifies the code duplication.

Reply:
Can you show an example? All we need to do is check if it matches a regex and then do the substitution, right? What length check do you think it's needed?

Reply:
I'm talking about this check. https://github.com/jitsi/jitsi-meet/pull/3551/files#diff-7646d8f5c54e9792a5ad011a8909ae91R44
But what you purpose we will be executing the regexp twice, which I don't like. Once for checking and once for replacing, is this right or I'm missing something?